pkg/daemon/server/service/pipeline_metrics_query.go

```diff
@@ -225,6 +225,13 @@ func (ps *pipelineMetadataQuery) GetVertexMetrics(ctx context.Context, req *daem
 		}
 	}
 
+	// in case any vertex queried for metrics is not running, this check prevents daemon crashes
+	for _, v := range metricsArr {
+		if v == nil {
+			return nil, fmt.Errorf("error while retrieving metrics for the %s vertex", req.GetVertex())
```

Comment:
this will break the UI; let's make sure the UI experience is not broken.

Reply:
I think UI should show no metrics available banner instead of crashing when not getting metrics

Reply:
Or even better UI can say the vertex pod is not running

Reply:
@vigith I think the UI experience is already broken when there is no metrics. This check will not change that

Reply:
that is correct, I have asked @veds-g to fix the UI experience too.

Reply:
Is this PR good to merge?

Reply:
i will be adding a UI fix in the same PR